celery/backends/base.py

```diff
@@ -647,11 +647,36 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.datetime.utcnow()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
-            'children': self.current_task_children(request),
```

Comment:
I would suggest to leave this here. By adding the `children` key conditionally in the metadata dictionary, you may be breaking some other functionality.

Reply:
children requires the request object anyway, so putting it behind the conditional should not harm anything, and may guard against issues (principally, test case mocks)

Reply:
I understand, however my point was that now the `children` key will be missing from the dictionary, unless `request` is not `None`. This could have unwanted side-effects, getting `KeyError` exceptions at other places or certain flows.

Reply:
I see your point, the function returns a None if request is none.  Reverted to original behavior.